test/runnable/cppa.d

```diff
@@ -983,6 +983,61 @@ void testeh3()
 }
 
 /****************************************/
+// 15579
+
+extern (C++)
+{
+    class Base
+    {
+        //~this() {}
+        void based() { }
+        ubyte x = 4;
+    }
+
+    interface Interface
+    {
+        int MethodCPP();
+        int MethodD();
+    }
+
+    class Derived : Base, Interface
+    {
+        short y = 5;
+        int MethodCPP();
+        int MethodD() { return 3; }
+    }
+
+    Derived cppfoo(Derived);
+    Interface cppfooi(Interface);
+}
+
+void test15579()
+{
+    Derived d = new Derived();
+    assert(d.x == 4);
+    assert(d.y == 5);
+    assert(d.MethodD() == 3);
+    assert(d.MethodCPP() == 30);
+
+    d = cppfoo(d);
+    assert(d.x == 7);
+    assert(d.y == 8);
+    version (Win64)
+    {
+        // needs more work
```

Comment:
> Does this suggest this is not yet compatible with MSC-Win64?
That's right. Win32 and Posix systems work fine, but VC++ is doing something different with the vtbl[]s. Grrr. The current fix is not wrong, it is still necessary for Win64. It's just that that particular test case still fails (before the PR, all of them failed).